src/main/java/vazkii/botania/common/item/equipment/tool/ItemStarSword.java

```diff
@@ -36,37 +40,49 @@ public ItemStarSword(Properties props) {
 	@Override
 	public void inventoryTick(ItemStack stack, Level world, Entity entity, int slot, boolean selected) {
 		super.inventoryTick(stack, world, entity, slot, selected);
-		if (entity instanceof Player player) {
-			MobEffectInstance haste = player.getEffect(MobEffects.DIG_SPEED);
-			float check = haste == null ? 0.16666667F : haste.getAmplifier() == 1 ? 0.5F : 0.4F;
+		if (!(entity instanceof Player player)) {
+			return;
+		}
 
-			if (player.getMainHandItem() == stack && player.attackAnim == check && !world.isClientSide) {
-				BlockHitResult pos = ToolCommons.raytraceFromEntity(player, 48, false);
-				if (pos.getType() == HitResult.Type.BLOCK) {
-					Vec3 posVec = Vec3.atLowerCornerOf(pos.getBlockPos());
-					Vec3 motVec = new Vec3((0.5 * Math.random() - 0.25) * 18, 24, (0.5 * Math.random() - 0.25) * 18);
-					posVec = posVec.add(motVec);
-					motVec = motVec.normalize().reverse().scale(1.5);
+		// Initialize timer for new items or items given to another player.
+		if (!ItemNBTHelper.verifyExistance(stack, STAR_CHARGED) || ItemNBTHelper.getInt(stack, STAR_CHARGED, 0) > player.tickCount) {
```

Comment:
is `player.tickCount` persistent or it restarts at 0 every time you load the world?

Reply:
I'm not sure about this one. Looks like it counts the number of ticks this entity has lived since creation, so I guess it will reset? 

But the code also checks if the value on the tag is greater than the bearer's tickCount, and it will always ensure the former is no greater than the latter, so the logic seems fine to me no matter what.